lib/improver/tests/utilities/test_DiscreteDifferenceBetweenAdjacentGridSquares.py

```diff
@@ -0,0 +1,223 @@
+        diff_array = np.array([[1, 2, 3],
+                               [3, 6, 9]])
+        time_coord = self.cube.coord('time')
+        proj_x_coord = self.cube.coord(axis='x')
+        result = self.plugin.create_difference_cube(
+            self.cube, "projection_y_coordinate", diff_array)
+        self.assertEqual(result.coord(axis='x'), proj_x_coord)
+        self.assertEqual(result.coord('time'), time_coord)
+
+
+class Test_calculate_difference(IrisTest):
+
+    """Test the calculate_difference method."""
+
+    def setUp(self):
+        """Set up cube."""
+        data = np.array([[1, 2, 3],
+                         [2, 4, 6],
+                         [5, 10, 15]])
+        self.cube = set_up_cube(data, "wind_speed", "m s-1")
+        self.plugin = DifferenceBetweenAdjacentGridSquares()
+
+    def test_x_dimension(self):
+        """Test differences calculated along the x dimension."""
+        expected = np.array([[1, 1],
+                             [2, 2],
+                             [5, 5]])
+        result = self.plugin.calculate_difference(self.cube, "x")
+        self.assertIsInstance(result, Cube)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_y_dimension(self):
+        """Test differences calculated along the y dimension."""
+        expected = np.array([[1, 2, 3],
+                             [3, 6, 9]])
+        result = self.plugin.calculate_difference(self.cube, "y")
+        self.assertIsInstance(result, Cube)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_missing_data(self):
+        """Test that the result is as expected when data is missing."""
+        data = np.array([[1, 2, 3],
+                         [np.nan, 4, 6],
+                         [5, 10, 15]])
+        cube = set_up_cube(data, "wind_speed", "m s-1")
+        expected = np.array([[np.nan, 2, 3],
+                             [np.nan, 6, 9]])
+        result = self.plugin.calculate_difference(cube, "y")
+        self.assertIsInstance(result, Cube)
+        self.assertArrayAlmostEqual(result.data, expected)
+
+    def test_masked_data(self):
+        """Test that the result is as expected when data is masked."""
+        data = ma.array([[1, 2, 3],
+                         [2, 4, 6],
+                         [5, 10, 15]],
+                        mask=[[0, 0, 0],
+                              [1, 0, 0],
+                              [0, 0, 0]])
+        cube = set_up_cube(data, "wind_speed", "m s-1")
+        expected = ma.array([[1, 2, 3],
+                             [3, 6, 9]],
+                            mask=[[1, 0, 0],
+                                  [1, 0, 0]])
+        result = self.plugin.calculate_difference(cube, "y")
+        self.assertIsInstance(result, Cube)
+        self.assertArrayEqual(result.data, expected)
+        self.assertArrayEqual(result.data.mask, expected.mask)
+
+
+class Test_process(IrisTest):
```

Comment:
At some point I think we should be checking the metadata is what we expect on our output cube